test/powershell/Language/Classes/Scripting.Classes.Attributes.Tests.ps1

```diff
@@ -216,3 +216,263 @@ Describe 'Type resolution with attributes' -Tag "CI" {
+
+        It 'Empty dynamically generated set throws in PowerShell script' {
+            $exc = {
+                Get-TestValidateSetPS5 -Param1 "TestString1" -ErrorAction Stop
+            } | ShouldBeErrorId "ParameterArgumentValidationError,Get-TestValidateSetPS5"
+            $exc.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId | Should BeExactly "ValidateSetGeneratedValidValuesListIsNull"
+        }
+
+        It 'Unimplemented valid values generator type throws in PowerShell script' {
+            {
+                Get-TestValidateSetPS6 -Param1 "AnyTestString" -ErrorAction Stop
+            } | ShouldBeErrorId "TypeNotFound"
+        }
+    }
+
+    Context 'CachedValidValuesGeneratorBase class tests' {
+
+        BeforeAll {
+            class GenValuesForParam : System.Management.Automation.CachedValidValuesGeneratorBase {
+                GenValuesForParam() : base(300) {
+                }
+
+                [String[]] GenerateValidValues() {
+
+                    return [string[]]("Test1","TestString1","Test2")
+                }
+            }
+
+            class GenValuesWithExpiration : System.Management.Automation.CachedValidValuesGeneratorBase {
+                GenValuesWithExpiration() : base(2) {
+                }
+
+                Static [bool] $temp = $true;
+
+                [String[]] GenerateValidValues() {
+
+                    if ([GenValuesWithExpiration]::temp) {
+                        [GenValuesWithExpiration]::temp = $false
+                        return [string[]]("Test1","TestString1","Test2")
+                    } else {
+                        [GenValuesWithExpiration]::temp = $true
+                        return [string[]]("Test1","TestString2","Test2")
+                    }
+                    
+                }
+            }
+
+
+            function Get-TestValidateSetPS4
+            {
+                [CmdletBinding()]
+                Param
+                (
+                    [ValidateSet([GenValuesForParam])]
+                    $Param1
+                )
+
+                $Param1
+            }
+
+            function Get-TestValidateSetPS5
+            {
+                [CmdletBinding()]
+                Param
+                (
+                    [ValidateSet([GenValuesWithExpiration])]
+                    $Param1
+                )
+
+                $Param1
+            }
+        }
+
+        It 'Can implement CachedValidValuesGeneratorBase in PowerShell' {
+            Get-TestValidateSetPS4 -Param1 "TestString1" -ErrorAction SilentlyContinue | Should BeExactly "TestString1"
+        }
+
+        It 'Can implement CachedValidValuesGeneratorBase with cache expiration in PowerShell' {
+            Get-TestValidateSetPS5 -Param1 "TestString1" -ErrorAction SilentlyContinue | Should BeExactly "TestString1"
+            Get-TestValidateSetPS5 -Param1 "TestString1" -ErrorAction SilentlyContinue | Should BeExactly "TestString1"
+            Start-Sleep 2

```

Comment:
I think the failure in CI build is due to this. You can either change to `base(1)`, or make it sleep for 3 seconds (not sure which is more reliable).